pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -73,959 +71,762 @@
-            return True
-        return self._map_devices_to_accelerator(_AcceleratorType.TPU)
-
-    @property
-    def use_tpu(self) -> bool:
-        return self._accelerator_type == _AcceleratorType.TPU and self.has_tpu
-
-    @property
-    def tpu_id(self) -> Optional[int]:
-        if self.use_tpu and isinstance(self.tpu_cores, list):
-            return self.tpu_cores[0]
-        return None
-
-    @property
-    def has_ipu(self) -> bool:
-        # Here, we are not checking for IPU availability, but instead if User has passed
-        # `ipus` to Trainer for training.
-        if self.ipus is not None or isinstance(self._strategy, IPUStrategy):
-            return True
-        return self._map_devices_to_accelerator(_AcceleratorType.IPU)
-
-    @property
-    def use_ipu(self) -> bool:
-        return self._accelerator_type == _AcceleratorType.IPU and self.has_ipu
-
-    def _set_devices_to_cpu_num_processes(self) -> None:
-        if self.num_processes == 1:
-            self._map_devices_to_accelerator(_AcceleratorType.CPU)
+    def _mapping_deprecated_devices_specfic_info_to_accelerator_and_device_flag(
+        self, devices, deprecated_devices_specific_flag, num_processes, gpus, ipus, tpu_cores
+    ):
+        # set devices base on num_processes, gpus, ipus, tpu_cores
+        if devices:
+            rank_zero_warn(
+                f"The flag `devices={devices}` will be ignored, "
+                f"instand the device specific number {deprecated_devices_specific_flag} will be used"
+            )
 
-    def _map_devices_to_accelerator(self, accelerator: str) -> bool:
-        if self.devices is None:
-            return False
-        if accelerator == _AcceleratorType.TPU and _TPU_AVAILABLE:
-            if self.devices == "auto":
-                self.devices = TPUAccelerator.auto_device_count()
-            self.tpu_cores = device_parser.parse_tpu_cores(self.devices)
-            return True
-        if accelerator == _AcceleratorType.IPU and _IPU_AVAILABLE:
-            if self.devices == "auto":
-                self.devices = IPUAccelerator.auto_device_count()
-            self.ipus = self.devices
-            return True
-        if accelerator == _AcceleratorType.GPU and torch.cuda.is_available():
-            if self.devices == "auto":
-                self.devices = GPUAccelerator.auto_device_count()
-            self.gpus = self.devices
-            self.parallel_device_ids = device_parser.parse_gpu_ids(self.devices)
-            return True
-        if accelerator == _AcceleratorType.CPU:
-            if self.devices == "auto":
-                self.devices = CPUAccelerator.auto_device_count()
-            if not isinstance(self.devices, int):
-                raise MisconfigurationException(
+        if [(num_processes is not None), (gpus is not None), (ipus is not None), (tpu_cores is not None)].count(
+            True
+        ) > 1:
+            rank_zero_warn("more than one device specifc flag has been set")
```

Comment:
```suggestion
            rank_zero_warn("more than one device specific flag has been set")
```
again we need to surface to users which of these flags are set to help them debug the warning